lib/db/mem.js

```diff
@@ -1052,9 +1087,31 @@ module.exports = function (log, error) {
       }
     })
 
+    // Sort emails so that primary email is first
+    userEmails.sort((a, b) => {
+      return (a.isPrimary === b.isPrimary) ? 0 : a.isPrimary ? -1 : 1
```

Comment:
Fwiw, I always struggle to read nested ternaries and prefer expanding them to full-blown `if` conditions. But if it needs to be a one-liner, doesn't `return b.isPrimary - a.isPrimary` produce the same result?